### PR TITLE
MC-13259 Multiple locations per deployment

### PR DESCRIPTION
**Pull request by danielvanacker**
#### Changes made
- Fixed formatting issues
- Added more fields for get and list workloads
- Modified Pop locations to show multiple locations

#### Related PRs
- [#42](https://github.com/cloudops/cloudmc-stackpath-plugin/pull/42)

### CODE REVIEW

File: source/includes/stackpath/_workloads.md [hunk 1]
```diff
@@ -52,16 +94,33 @@ Attributes | &nbsp;
...
 `type`<br/>*string* | Specify whether a workload is a VM-based workload or container-based.
 `network`<br/>*string* | Network interfaces to bind to the workload's instances.
+`specs`<br/>*string* | Specification type for resources which are allocated to each instance in a workload.
 `cpu`<br/>*string* | The number of vCPUs for the workload's instance.
 `memory`<br/>*string* | The memory size for the workload's instance.
-`isRemoteManagementEnabled`<br/>*boolean* | Specify if remote management is enabled on workload instance or not.
+`isRemoteManagementEnabled`<br/>*boolean* | Specifies if remote management is enabled on workload instance or not.
 `image`<br/>*string* | The workload's instance operating system image.
+`addImagePullCredentialsOption`<br/>*boolean* | It is used to indicate if additional credentials to pull container image are provided or not. Only applicable to workloads of `type` 'CONTAINER'.
+`containerUsername` <br/>*string* | The username used to authenticate the image pull. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'True'.
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]:
```suggestion
`containerUsername` <br/>*string* | The username used to authenticate the image pull. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'true'.
```

`True` is not a valid boolean value for JSON.

File: source/includes/stackpath/_workloads.md [hunk 2]
```diff
@@ -52,16 +94,33 @@ Attributes | &nbsp;
...
+`containerServer` <br/>*string* | The server that the credentials should be used with. This value will default to the docker hub registry when not set. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'True'.
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]: same as above

File: source/includes/stackpath/_workloads.md [hunk 3]
```diff
@@ -52,16 +94,33 @@ Attributes | &nbsp;
...
+`containerEmail` <br/>*string* | The email address to use for the docker registry account. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'True'.
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]: same as above

Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: ed6000d

File: source/includes/stackpath/_workloads.md (hunk 2 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: ed6000d

File: source/includes/stackpath/_workloads.md (hunk 1 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: ed6000d

File: source/includes/stackpath/_workloads.md [hunk 4]
```diff
@@ -52,16 +94,33 @@ Attributes | &nbsp;
...
+`containerEmail` <br/>*string* | The email address to use for the docker registry account. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'true'.
+`environmentVariableKey` <br/>*string* | The location to obtain a value for an environment variable. Only applicable to workloads of `type` 'CONTAINER'.
+`environmentVariableValue` <br/>*string* | An environment variable's value. Only applicable to workloads of `type` 'CONTAINER'.
+`secretEnvironmentVariableKey` <br/>*string* | The location to obtain a value for a secret environment variable. Only applicable to workloads of `type` 'CONTAINER'.
+`firstBootSshKey`<br/>*string* | The ssh key for the VM image. Only applicable to workloads of `type` 'VM'.
+`persistenceStoragePath`<br/>*string* | The path in an instance to mount a volume.
+`persistenceStorageSize`<br/>*int* | The size of the mounted volume (in GB).
+`deploymentName`<br/>*string* | The name of the deployment.
+`deploymentPops`<br/>*array* | The points of presence of a deployment. In the format [A-Z][A-Z][A-Z].
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]:
```suggestion
`deploymentPops`<br/>*array* | The points of presence of a deployment. In the regex format `[A-Z][A-Z][A-Z]`.
```

File: source/includes/stackpath/_workloads.md [hunk 5]
```diff
@@ -106,16 +175,33 @@ Attributes | &nbsp;
...
 `type`<br/>*string* | Specify whether a workload is a VM-based workload or container-based.
 `network`<br/>*string* | Network interfaces to bind to the workload's instances.
+`specs`<br/>*string* | Specification type for resources which are allocated to each instance in a workload.
 `cpu`<br/>*string* | The number of vCPUs for the workload's instance.
 `memory`<br/>*string* | The memory size for the workload's instance.
-`isRemoteManagementEnabled`<br/>*boolean* | Specify if remote management is enabled on workload instance or not.
+`isRemoteManagementEnabled`<br/>*boolean* | Specifies if remote management is enabled on workload instance or not.
 `image`<br/>*string* | The workload's instance operating system image.
+`addImagePullCredentialsOption`<br/>*boolean* | It is used to indicate if additional credentials to pull container image are provided or not. Only applicable to workloads of `type` 'CONTAINER'.
+`containerUsername` <br/>*string* | The username used to authenticate the image pull. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'True'.
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]:
```suggestion
`containerUsername` <br/>*string* | The username used to authenticate the image pull. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'true'.
```

File: source/includes/stackpath/_workloads.md [hunk 6]
```diff
@@ -106,16 +175,33 @@ Attributes | &nbsp;
...
+`containerServer` <br/>*string* | The server that the credentials should be used with. This value will default to the docker hub registry when not set. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'True'.
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]:
```suggestion
`containerServer` <br/>*string* | The server that the credentials should be used with. This value will default to the docker hub registry when not set. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'true'.
```

File: source/includes/stackpath/_workloads.md [hunk 7]
```diff
@@ -106,16 +175,33 @@ Attributes | &nbsp;
...
+`containerEmail` <br/>*string* | The email address to use for the docker registry account. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'True'.
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]:
```suggestion
`containerEmail` <br/>*string* | The email address to use for the docker registry account. Only applicable to workloads of `type` 'CONTAINER' and `addImagePullCredentialsOption` is 'true'.
```

File: source/includes/stackpath/_workloads.md [hunk 8]
```diff
@@ -126,7 +212,9 @@ curl -X POST \
     -d "request_body" \
     "https://cloudmc_endpoint/api/v1/services/stackpath/test-area/workloads"
 ```
+<<<<<<< HEAD
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]: 😱

File: source/includes/stackpath/_workloads.md [hunk 9]
```diff
@@ -203,8 +292,11 @@ Required | &nbsp;
  `firstBootSshKey(s)`<br/>*string* | If creating a VM-based workload, SSH keys are required. Multiple SSH keys can be separated by commas.
  `specs`<br/>*string* | Specification type for resources which are allocated to each instance in a workload. Supported specifications are `SP-1 (1 vCPU, 2 GB RAM)`,`SP-2 (2 vCPU, 4 GB RAM)`,`SP-3 (2 vCPU, 8GB RAM)`,`SP-4 (4 vCPU, 16 GB RAM)`,`SP-5 (8 vCPU, 32 GB RAM)`.
  `deploymentName`<br/>*string* | The name of the deployment.
- `deploymentPops`<br/>*string* | The point of presence of a deployment. In the format [A-Z][A-Z][A-Z].
- `enableAutoScaling` <br/>*boolean* | Specify if you would like to enable autoscaling. If enabled, then `cpuUtilization` , `minInstancesPerPop` and `maxInstancesPerPop` are required else `deploymentInstancePerPops` is required.
+ `deploymentPops`<br/>*array* | The points of presence of a deployment. In the format [A-Z][A-Z][A-Z].
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]:
```suggestion
 `deploymentPops`<br/>*array* | The points of presence of a deployment. In the regex format `[A-Z][A-Z][A-Z]`.
```

File: source/includes/stackpath/_workloads.md [hunk 10]
```diff
@@ -287,29 +373,34 @@ curl -X PUT \
...
-`slug`<br/>*string* | A workload's programmatic name. Workload slugs are used to build its instances names.
 `type`<br/>*string* | Specify whether a workload is a VM-based workload or container-based. Can be either 'VM' or 'CONTAINER'.
-`image`<br/>*string* | The location of a Docker image to run as a container. Only available when `type`is equal to 'CONTAINER'.
+`image`<br/>*string* | The location of a Docker image to run as a container. Only editable when `type`is equal to 'CONTAINER'.
 `cpu`<br/>*string* | The number of vCPUs for the workload's instance.
 `memory`<br/>*string* | The memory size for the workload's instance.
 `specs`<br/>*string* | Specification type for resources which are allocated to each instance in a workload.
 `deploymentName`<br/>*string* | The name of the deployment.
-`deploymentPops`<br/>*string* | The point of presence of a deployment. In the format [A-Z][A-Z][A-Z].
+`deploymentPops`<br/>*string* | The points of presence of a deployment. In the format [A-Z][A-Z][A-Z].
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]: same as above

**Review by y0geshdev — COMMENTED** [review]
Some suggestions and small typos.

File: source/includes/stackpath/_workloads.md (hunk 8 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: 😮

File: source/includes/stackpath/_workloads.md (hunk 10 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: 28be41d

File: source/includes/stackpath/_workloads.md (hunk 9 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: 28be41d

File: source/includes/stackpath/_workloads.md (hunk 8 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: 28be41d

File: source/includes/stackpath/_workloads.md (hunk 7 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: 28be41d

File: source/includes/stackpath/_workloads.md (hunk 6 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: 28be41d

File: source/includes/stackpath/_workloads.md (hunk 5 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: 28be41d

File: source/includes/stackpath/_workloads.md (hunk 4 above)
Comment (danielvanacker) [reply source/includes/stackpath/_workloads.md]: 28be41d

**Review by y0geshdev — APPROVED** [review]
lgtm. Thanks for the changes. 🙏

**Review by razine-bensari — APPROVED** [review]
lgtm! :)

**Review by RGitB — APPROVED** [review]
lgtm too :D

**Pull request merged**